edb/schema/functions.py

```diff
@@ -1122,22 +1125,30 @@ def compile_function(
             inlined_defaults=has_inlined_defaults,
         )
 
+        # retain the compiler options context to later extract
+        # dml_exprs from it
+        options = qlcompiler.CompilerOptions(
+            anchors=param_anchors,
+            func_params=params,
+            # the body of a session_only function can contain calls to
+            # other session_only functions
+            session_mode=session_only,
+        )
         compiled = type(body).compiled(
             body,
             schema,
-            options=qlcompiler.CompilerOptions(
-                anchors=param_anchors,
-                func_params=params,
-                # the body of a session_only function can contain calls to
-                # other session_only functions
-                session_mode=session_only,
-            ),
+            options=options,
         )
 
         ir = compiled.irast
         assert isinstance(ir, irast.Statement)
         schema = ir.schema
 
+        if options.dml_exprs:
```

Comment:
Please use `irast.Statement` to communicate the characteristics of IR.

Reply:
OK, I'll move that from the context options to `irast.Statement`.